app/models/unit_activity_set.rb

```diff
@@ -10,4 +10,8 @@ class UnitActivitySet < ActiveRecord::Base
   # This ensures that id provided is also valid, so there exists an unit with that id
   validates :activity_type, presence: true
   validates :unit,          presence: true
+
+  # Always add a unique index to the DB to prevent new records from passing the validations when checked at the same time before being written
+  # For reference, see unique index migrations of unit activity sets
+  validates_uniqueness_of :activity_type, :scope => :unit, message: 'already exists for the unit'
```

Comment:
You can have multiple activity sets of the same type - multiple tutorials for example. Tute1 and Tute2